Science cyborg chassis

neurovermi opened the pull request:
<!-- IT'S NOT WIZDENS REPO, IF YOU WANT TO ADD YOUR CHANGES ON ALL SERVERS, CREATE PR TO WIZDENS REPO -->

## Short description
<!-- What do you propose to change with your PR? -->
This adds a science cyborg chassis variant. Now instead of the generic borg, science gets its very own designated helper. The generic cyborg now only uses generic modules, as intended.

## Why we need to add this
<!-- What is the reason for adding these changes? Please post links to Discussions as well as Bug Reports here. Please describe how this will change the game balance. -->
This was always the intent, as can be seen in the YAML, but wizden lacked the sprites. As a result, generic cyborgs ended up on double duty, which never really made sense gameplay or RP wise. A Starlight user created some very good ones, so now we can have our very own science borg, allowing us to cleanly separate the roles.

## Media (Video/Screenshots)
<!--
If your PR contains in-game changes you must provide screenshots/videos of the changes.
-->

https://github.com/user-attachments/assets/9f954e2d-ab24-4ecc-8a33-778da8e0e76f



## Checks
<!-- check boxes for faster reviewing of your PR -->

- [x] I do not require assistance to complete the PR.
- [x] Before posting/requesting review of a PR, I have verified that the changes work.
- [x] I have added screenshots/videos of the changes, or this PR does not change in-game mechanics.
- [x] I affirm that my changes are licensed under the [Starlight Fork License](https://github.com/ss14Starlight/space-station-14/blob/Starlight/LICENSE-Starlight.TXT) and grant permission for use in this repository under its conditions.

**Changelog**
<!--
If you want the players to know about changes made in this PR, specify them using the template outside the comment. Short and informative.
-->
:cl: neurovermi
- add: A new cyborg chassis! Science now has a dedicated cyborg. Sprites made by piksqu
- tweak: Generic cyborgs have been fired from R&D. They now use only generic modules and common radio as intended.

neurovermi committed:
Create initial files

Structured the chassis RSIs a bit differently from wizden to make it less monolithic

neurovermi committed:
Sync animations and delete chassis subfolder

I don't like how the switchable borg system is set up. The sprites might have to go into the chassis.rsi as a result.

neurovermi committed:
Add science borg sprites

Due to how these were setup they did have to go in the existing chassis.rsi. Don't like that. Worth refactoring.

neurovermi committed:
Proto changes (no loc text yet)

neurovermi committed:
locale files

neurovermi committed:
Remove anim pausing on death and rename sync function

neurovermi committed:
Rename function, add docstring, remove unnecessary method

Elected to leave unused LayerGetAnimationTime overloads for future use. Don't think it'll hurt.

neurovermi commented:
Note for reviewers/maintainers: Due to the way in which borg switchable chassis were set up by wizden, any new borg sprites *had* to go in the existing chassis.rsi for it to function. Very not so good way they did this, I don't like it and I can't even leave comments in the Json since its well... Json.

ShadesMars commented on Resources/Prototypes/_StarLight/Entities/Mobs/Cyborgs/borg_chassis.yml:
Add a OptionsVisualizer comp / Support for ReducedMotion where sprites are static.

neurovermi replied:
Re-review requested, see AnimationSyncSystem.cs

neurovermi committed:
Add missing Starlight comment

neurovermi commented:
Going to mark this as a draft until I can decide on a good way to handle the requested ReducedMotion support

neurovermi committed:
Client-side component for synchronizing spriteframes to a given layer

Component isn't registering for some reason despite the attribute.

neurovermi committed:
Add ReducedMotion support for synced animated sprites

neurovermi committed:
Cleanup PR, remove unnecessary whitespace and yml changes

neurovermi commented:
To be completely honest - This solution is not great, and it's entirely a result of the way borgs are set up. To that end, I'll be opening an issue for a refactor detailing some of the issues I ran into while creating the PR.

neurovermi committed:
Style and function accessors

Really, really hate having to put ReducedMotion here instead of just using an OptionsVisualizer - but that's the borg system for you.

ShadesMars commented:
This PR has become stale, however with the new borg system, this should be worked on.

ShadesMars commented:
PR is now "Stale" however, feel free to re-open this PR if you wish; however i would love with your permission to add science borgs with the new borg selection system, by yourself or i do it when i have free time.
